sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverClientIntegrationTest.java

```diff
@@ -0,0 +1,578 @@
+        ServiceBusReceivedMessage receivedMessage = asList.get(0);
+
+        assertNotNull(receivedMessage);
+
+        // Act
+        ServiceBusReceivedMessage receivedPeekMessage = receiver.peekAt(receivedMessage.getSequenceNumber());
+
+        // Assert
+        assertMessageEquals(receivedPeekMessage, messageId, isSessionEnabled);
+    }
+
+    /**
+     * Verifies that we can send and peek a batch of messages.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void peekBatchMessages(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, isSessionEnabled);
+        final int maxMessages = 2;
+
+        sendMessage(message);
+        sendMessage(message);
+
+        // Act
+        IterableStream<ServiceBusReceivedMessage> iterableMessages = receiver.peekBatch(maxMessages);
+
+        // Assert
+        Assertions.assertEquals(maxMessages, iterableMessages.stream().collect(Collectors.toList()).size());
+    }
+
+    /**
+     * Verifies that we can send and peek a batch of messages.
+     */
+    @MethodSource("messagingEntityProvider")
+    @ParameterizedTest
+    void peekBatchMessagesFromSequence(MessagingEntityType entityType) {
+        // Arrange
+        setSenderAndReceiver(entityType, false);
+
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, false);
+        final int maxMessages = 2;
+        final int fromSequenceNumber = 1;
+
+        sendMessage(message);
+        sendMessage(message);
+
+        // Act
+        IterableStream<ServiceBusReceivedMessage> iterableMessages = receiver.peekBatchAt(maxMessages, fromSequenceNumber);
+
+        // Assert
+        Assertions.assertEquals(maxMessages, iterableMessages.stream().collect(Collectors.toList()).size());
+    }
+
+    /**
+     * Verifies that we can dead-letter a message.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void deadLetterMessage(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, isSessionEnabled);
+
+        sendMessage(message);
+
+        final IterableStream<ServiceBusReceivedMessage> messageIte = receiver.receive(1, TIMEOUT);
+        Assertions.assertNotNull(messageIte);
+
+        final List<ServiceBusReceivedMessage> asList = messageIte.stream().collect(Collectors.toList());
+        ServiceBusReceivedMessage receivedMessage = asList.get(0);
+
+        assertNotNull(receivedMessage);
```

Comment:
Same with the comment above. If you did .get(0) and the list was empty, it would throw. This check is incorrect.

Reply:
And all other instances of it.